mlir/lib/Conversion/LLVMCommon/PrintCallHelper.cpp

```diff
@@ -59,8 +59,13 @@ void mlir::LLVM::createPrintStrCall(
   SmallVector<LLVM::GEPArg> indices(1, 0);
   Value gep =
       builder.create<LLVM::GEPOp>(loc, ptrTy, arrayTy, msgAddr, indices);
-  Operation *printer =
-      LLVM::lookupOrCreatePrintStringFn(moduleOp, runtimeFunctionName);
-  builder.create<LLVM::CallOp>(loc, TypeRange(), SymbolRefAttr::get(printer),
-                               gep);
+  if (auto printer =
+          LLVM::lookupOrCreatePrintStringFn(moduleOp, runtimeFunctionName);
+      succeeded(printer)) {
+    builder.create<LLVM::CallOp>(loc, TypeRange(),
+                                 SymbolRefAttr::get(printer.value()), gep);
+  } else {
+    return failure();
+  }
+  return success();
```

Comment:
```suggestion
  FailureOr<LLVM::LLVMFuncOp> printer =
          LLVM::lookupOrCreatePrintStringFn(moduleOp, runtimeFunctionName);
  if(failed(printer))
    return failure();
  builder.create<LLVM::CallOp>(loc, TypeRange(),
                                 SymbolRefAttr::get(printer.value()), gep);
  return success();
```
Nit: This might be a bit nicer